### PR TITLE
[PDP-1094] Refactored average_end_of_day_balance and sum_of_minimum_balances

### PR DESCRIPTION
**Pull request by AdityaRao1996**
- Refactored `average_end_of_day_balance` and `sum_of_minimum_balances` features to make use of `pngme-api`'s `AsyncClient` in order to make the API calls in an asynchronous manner

Request @pngme/data-science to please review the changes to make sure that the changes make sense

### CODE REVIEW

File: lib/sum_of_minimum_balances/main.py [hunk 1]
```diff
@@ -1,97 +1,124 @@
...
 
     Args:
-        api_client: Pngme API client
-        user_uuid: the Pngme user_uuid for the mobile phone user
-        utc_starttime: the datetime for the left-hand-side of the time-window
-        utc_endtime: the datetime for the right-hand-side of the time-window
+        api_client: Pngme Async API client
+        user_uuid: Pngme mobile phone user_uuid
+        utc_time: the time at which the balances are computed
+        cutoff_interval: if balance hasn't been updated within this interval, then balance record is stale, and method returns None
```
Comment (kevindingpngme) [inline lib/sum_of_minimum_balances/main.py:23]: We should explain the definition of the argument, and then talk about edge case.

File: lib/average_end_of_day_balance/main.py [hunk 2]
```diff
@@ -1,32 +1,34 @@
...
     Args:
-        client: Pngme API client
+        api_client: Pngme Async API client
         user_uuid: the Pngme user_uuid for the mobile phone user
-        utc_starttime: the datetime for the left-hand-side of the time-window
-        utc_endtime: the datetime for the right-hand-side of the time-window
+        utc_time: the time-zero to use in constructing the 0-30, 31-60 and 61-90 windows
 
     Returns:
         the average end-of-day total balance for a given time window
```
[the remark this answers is not in the document]
Comment (nickmasson) [reply lib/average_end_of_day_balance/main.py:29]: +1

File: lib/average_end_of_day_balance/main.py [hunk 3]
```diff
@@ -41,34 +43,35 @@ def get_average_end_of_day_balance(
...
+            institution_name=institutions_w_depository[index].institution_id
         )
         balance_df_list.append(institution_balances_df)
 
     # 1. Combine all institution balances into single dataframe
-    balances_df = pd.concat(balance_df_list)
-    if balances_df.empty:
-        return None
+    if balance_df_list:
+        balances_df = pd.concat(balance_df_list)
```
Comment (nickmasson) [inline lib/average_end_of_day_balance/main.py:72]: len(balance_df) can still be zero here, if `balance_df_list` is a list of empty dataframes. we need to check for that.

**Review by nickmasson — APPROVED** [review]
we need to apply an additional check to avoid bugginess. rest of the logic looks ok from what I can tell. I'd like to refactor the avg_eod_balance to make it more pandasian (pandastic? pandalicious?) at some point, but will leave that for future improv.

**Pull request merged**